### PR TITLE
Update documentation for Physics2DServer

### PR DESCRIPTION
**Pull request by seenloitering**
Undocumented functions have been added and several existing explanations have been clarified or fleshed out. A more extensive description has been added including a (maybe too large) code sample that demonstrates some typical usage.

instance_id and canvas_id functions remain insufficiently explained imo.

### CODE REVIEW

**Comment by YeldhamDev** [comment]
Inside `codeblock`, 4 spaces are used for indentation.

**Comment by YeldhamDev** [comment]
It should be written like this:
```
<tab>Documentation text.
<tab>[codeblock]
<tab>Code.
<tab><4 spaces>More code.
<tab>[/codeblock]
```

**Comment by seenloitering** [comment]
Ah. I see! I'll change it.

File: doc/classes/Physics2DServer.xml [hunk 1]
```diff
@@ -1028,14 +1128,23 @@
...
+				[b]SHAPE_RAY[/b]: [method shape_set_data] [codeblock]( RID shape, {”length”: float length, “slips_on_slope”: bool} )[/codeblock]
+				[b]SHAPE_SEGMENT[/b]: [method shape_set_data] [codeblock]( RID shape,array[float pointA.x, float pointA.y, float pointB.x, float pointB.y] )[/codeblock]
+				[b]SHAPE_CIRCLE[/b]: [method shape_set_data] [codeblock]( RID shape, float radius)[/codeblock]
+				[b]SHAPE_RECTANGLE[/b]: [method shape_set_data] [codeblock]( RID shape, array[float width, float height)] )[/codeblock]
+				[b]SHAPE_CAPSULE[/b]: [method shape_set_data] [codeblock]( RID shape, array[float height, float width] )[/codeblock]
+				[b]SHAPE_CONVEX_POLYGON[/b]: [method shape_set_data] [codeblock]( RID shape, array[Vector2(), Vector2(), ...] )[/codeblock]
+				[b]SHAPE_CONCAVE_POLYGON[/b]: [method shape_set_data] [codeblock]( RID shape, array[Vector2(), Vector2(), ...] )[/codeblock]
+				Note: SHAPE_CONVEX_POLYGON is the same type used by CollisionPolygon2D when it is set to “solids,” and SHAPE_CONCAVE_POLYGON  is used when CollisionPolygon2D is set to “segments.”
```
Comment (YuriSizov) [inline doc/classes/Physics2DServer.xml:1140]: Those `SHAPE_*` strings should be references to the appropriate constants. Also, inline code is formatted with `[code]` not `[codeblock]`.

File: doc/classes/Physics2DServer.xml [hunk 2]
```diff
@@ -4,7 +4,91 @@
...
+		Godot assigns an [code]object_instance_id[/code] to a Node2D when it is instantiated. This id is unique for each object in the node tree. Because Physics2DServer does not instantiate classes as [code]Node2D[/code] objects, these resources do not have valid object instance IDs and are not suitable for attachment to the node tree. Instead, they default to zero.
+		As with object instance IDs, Physics2DServer does not create resources with a valid [code]canvas_instance_id[/code]. Instead, the canvas ID will be zero. The canvas is the same as is used by a [code]Canvaslayer[/code].
+		There is currently no supported method for casting areas or bodies created with Physics2DServer as Node2D objects. Because of this, you’re better off instantiating nodes from Node2D.new() or one of Node2D’s member classes, then using get_rid() to retrieve an RID Physics2DServer can work with.
+		Attach the following script to a Node2D in a new Godot project and run it. It demonstrates many of the basic relationships between Physics2DServer functions.
+	[codeblock]
+	extends Node2D
+
+	var aFooBody:RID
+	var aBarBody:RID
+	var gravityWell:RID
```
Comment (YuriSizov) [inline doc/classes/Physics2DServer.xml:22]: We shouldn't put such giant codeblocks in the docs. It also needs a C# counterpart. From the looks of it, this description can be made into a dedicated tutorial article and just linked here.

**Comment by YuriSizov** [comment]
@seenloitering Are you willing to continue working on this? It needs to be rebased and cleaned up. The huge example needs to be moved elsewhere, I think. The rest looks mostly fine, and we definitely want to have such documentation.

**Comment by akien-mga** [comment]
Superseded by #70723.

**Pull request closed without merging**